### PR TITLE
feat: Turn on Snuba for the project details endpoint

### PR DESCRIPTION
**Pull request by lynnagara**
Fixes an issue where latest/oldest event found by Snuba is not available in legacy.

Ref: SEN-401, JAVASCRIPT-5VE

### CODE REVIEW

File: tests/snuba/api/endpoints/test_group_events_oldest.py [hunk 1]
```diff
@@ -41,21 +41,18 @@ def setUp(self):
 
         self.group = Group.objects.first()
 
-    def test_simple(self):
+    def test_snuba_no_environment(self):
+        options.set('snuba.events-queries.enabled', True)
```
Comment (markstory) [inline tests/snuba/api/endpoints/test_group_events_oldest.py]: Were these tests even hitting snuba before?

File: src/sentry/db/models/manager.py [hunk 2]
```diff
@@ -340,6 +340,8 @@ def from_event_id(self, id_or_event_id, project_id):
 
             id_or_event_id = event.event_id
 
+            project_id = event.project_id
```
Comment (wedamija) [inline src/sentry/db/models/manager.py]: Why do we need to do this?

File: tests/snuba/api/endpoints/test_group_events_oldest.py (hunk 1 above)
Comment (wedamija) [reply tests/snuba/api/endpoints/test_group_events_oldest.py]: `test_snuba_no_environment` was basically doing this before

**Review by wedamija — APPROVED** [review]

File: src/sentry/db/models/manager.py (hunk 2 above)
Comment (lynnagara) [reply src/sentry/db/models/manager.py]: Oh yeah I guess we don't need this. I was kinda thinking we could have this function also handle the case where it's being called with a event ID as database key and no project, but that might be unnecessary

Comment (lynnagara) [reply src/sentry/db/models/manager.py]: Removing

Comment (wedamija) [reply src/sentry/db/models/manager.py]: Are we actually able to fetch an event without a project id? I think we shard on project, so I'm not sure there's a good way to do so?

Comment (lynnagara) [reply src/sentry/db/models/manager.py]: Yeah we should probably just never call this without a project ID

**Pull request merged**